controllers/progressivesync_controller.go

```diff
@@ -267,6 +280,67 @@ func (r *ProgressiveSyncReconciler) getClustersFromSelector(ctx context.Context,
 	return secrets, nil
 }
 
+// Calculate the hash value from the the source ref object (Spec of the ApplicationSet)
+// Store hash value inside the progressive sync annotation map
+
+func (r *ProgressiveSyncReconciler) updatePsHashAnnotation(ctx context.Context, pr *syncv1alpha1.ProgressiveSync, newHashValue string) error {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		key := client.ObjectKeyFromObject(pr)
+		latest := syncv1alpha1.ProgressiveSync{}
+		if err := r.Client.Get(ctx, key, &latest); err != nil {
+			return err
+		}
+
+		if len(latest.Annotations) == 0 {
+			latest.Annotations = make(map[string]string)
+		}
+
+		val := latest.Annotations["specHashValue"]
+		if val != newHashValue {
+			r.Log.Info("Setting the annotation for the new hash value")
+			latest.Annotations["specHashValue"] = newHashValue
+			if err := r.Client.Update(ctx, &latest); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	return retryErr
+}
+
+func (r *ProgressiveSyncReconciler) calculateHashValue(ctx context.Context, pr *syncv1alpha1.ProgressiveSync) (string, error) {
+	key := client.ObjectKeyFromObject(pr)
+	latest := syncv1alpha1.ProgressiveSync{}
+	if err := r.Client.Get(ctx, key, &latest); err != nil {
+		return "", err
+	}
+
+	////////////////////////////////////////////////
```

Comment:
we prob don't need those comments?

Reply:
Yes, sorry, the PR is still WIP. I think we will want to actually use what's commented as it is more generic and not specific to the ApplicationSet. But I was still working on it, I decided to go with the direct approach of using ApplicationSet for now.

Reply:
This is now removed.